pkg/webhook/resources/addon/validator.go

```diff
@@ -83,12 +83,26 @@ func validateUpdatedAddon(newAddon *v1beta1.Addon, oldAddon *v1beta1.Addon) erro
 		return werror.NewBadRequest("chart field cannot be changed.")
 	}
 
+	if v1beta1.AddonOperationInProgress.IsTrue(oldAddon) {
```

Comment:
If any operation fails the addon will be marked failed. enable/disable/deletion should be possible then. 

The aim is to avoid in-flight updates when a helm chart may already be in a install/upgrade phase.